### ISSUE TITLE
ICE: Throw in ArraySliceType::copyForLocation(DataLocation, bool) 

### ISSUE THREAD

**Issue by agroce** [ethereum/solidity#8383]
## Description

This contract:
```solidity
contract c{
  function i()public{
    int a;
    byte[]memory b;
    byte[]memory k=true?a:b;
    [//
     [1][:]]+
      a;b;
  }
}
```

produces:

```
Internal compiler error during compilation:
/home/user/solidity/libsolidity/ast/Types.h(797): Throw in function virtual std::unique_ptr<ReferenceType> solidity::frontend::ArraySliceType::copyForLocation(solidity::frontend::DataLocation, bool) const
Dynamic exception type: boost::exception_detail::clone_impl<solidity::langutil::InternalCompilerError>
std::exception::what: 
[solidity::util::tag_comment*] = 
```

when compiled with solc

On master, using AFL fuzzing.  First one I've seen from normal AFL, but not (yet at least) from the mutant-based version.

## Environment

- Compiler version:  0.6.3-develop.2020.2.10+commit.e8eb1f2d.Linux.clang
- Target EVM version (as per compiler settings): N/A
- Framework/IDE (e.g. Truffle or Remix): N/A
- EVM execution environment / backend / blockchain client: N/A
- Operating system: Ubuntu 18.04 in docker

## Steps to Reproduce

Above shows pretty clearly, I think.

**Comment by agroce** [comment]
If it's any help, this maybe simpler code does the same thing:
```
contract C {
    function f(int56[] calldata) external {
	int56 = [2][:].a;
    }
} 
```

**Comment by bshastry** [comment]
This may be a duplicate of #8265 going by location of the failed assertion.

**Comment by agroce** [comment]
Yes, looks likely, must've made an error in my search for the message

**Comment by axic** [comment]
Still reproducible today.

**Comment by bshastry** [comment]
This issue is no longer reproducible, possibly fixed by #9027
